From e182f96c7805047715cd3b7c083bd21dfc25faa0 Mon Sep 17 00:00:00 2001
From: Preeti Sachan <preeti.sachan@intel.com>
Date: Wed, 9 Mar 2022 11:00:58 -0500
Subject: [PATCH] fix compiling with efivar version 38

Refer efivar v38, source file "src/include/efivar/efivar-types.h" at
line no. 81 and "src/include/efivar/efivar-dp.h" at line no. 1199.

Signed-off-by: Preeti Sachan <preeti.sachan@intel.com>
---
 linux/libfwup.c |  2 +-
 linux/util.h    | 15 +--------------
 2 files changed, 2 insertions(+), 15 deletions(-)

diff --git a/linux/libfwup.c b/linux/libfwup.c
index 0352ab8..185352d 100644
--- a/linux/libfwup.c
+++ b/linux/libfwup.c
@@ -2103,7 +2103,7 @@ fwup_print_update_info(void)
 		efi_guid_t guid = info->guid;
 		char *id_guid = NULL;
 		ssize_t dp_sz;
-		char *path;
+		unsigned char *path;
 
 		rc = efi_guid_to_id_guid(&guid, &id_guid);
 		if (rc < 0)
diff --git a/linux/util.h b/linux/util.h
index a761d75..125bcd5 100644
--- a/linux/util.h
+++ b/linux/util.h
@@ -16,6 +16,7 @@
 #include <limits.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <efivar/efivar.h>
 
 #define  _(String) gettext (String)
 #define Q_(String) dgettext (NULL, String)
@@ -320,20 +321,6 @@ untilt_slashes(char *s)
 	return s;
 }
 
-typedef struct {
-	uint16_t year;
-	uint8_t month;
-	uint8_t day;
-	uint8_t hour;
-	uint8_t minute;
-	uint8_t second;
-	uint8_t pad1;
-	uint32_t nanosecond;
-	uint16_t timezone;
-	uint8_t daylight;
-	uint8_t pad2;
-} efi_time_t;
-
 #define CAPSULE_FLAGS_PERSIST_ACROSS_RESET    0x00010000
 #define CAPSULE_FLAGS_POPULATE_SYSTEM_TABLE   0x00020000
 #define CAPSULE_FLAGS_INITIATE_RESET          0x00040000
-- 
2.25.1

